### PR TITLE
Added update command

### PR DESCRIPTION
**Pull request by petarjuki7**
App update command

### CODE REVIEW

**Comment by MatejVukosav** [comment]
Can you add to docs how to use it?

**Comment by petarjuki7** [comment]
`cargo run -p meroctl -- --node-name <node_name> context update -a  <app_id> --context-id <contrext_id>  -p <member_public_key> --path <path_to_app>`

**Review by miraclx — COMMENTED** [review]
so.. we need the following:

```console
$ meroctl context update --context-id <context-id> -a <application-id> --as <identity>
# one-off update

$ meroctl context update --context-id <context-id> --path ./path/to/file --as <identity>
# one-off update

$ meroctl context update --context-id <context-id> --path ./path/to/file --watch --as <identity>
# recurring update
```

see the `call` command for how identity is managed

`<application-id>` and `<path>` are mutually exclusive, and `--watch` should only be allowed when `--path` is present.

File: crates/meroctl/src/cli/context/update.rs [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+    #[clap(
+        long,
+        conflicts_with = "application_id",
+        help = "Path to the application file to watch and install locally"
+    )]
+    path: Option<Utf8PathBuf>,
+
+    #[clap(long, help = "Metadata needed for the application installation")]
+    metadata: Option<String>,
```
Comment (miraclx) [inline crates/meroctl/src/cli/context/update.rs]: this should also conflict with `application_id`

File: crates/meroctl/src/cli/context/update.rs [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+                    &config.identity,
+                    executor_public_key,
+                )
+                .await?;
+            }
+            Self {
+                context_id,
+                application_id: None,
+                path: Some(path),
+                metadata: None,
```
Comment (miraclx) [inline crates/meroctl/src/cli/context/update.rs]:
```suggestion
                metadata,
```

File: crates/meroctl/src/cli/context/update.rs [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+                    &client,
+                    multiaddr,
+                    context_id,
+                    path,
+                    metadata.map(String::into_bytes),
+                    &config.identity,
+                    executor_public_key,
+                )
+                .await?;
+            }
```
Comment (miraclx) [inline crates/meroctl/src/cli/context/update.rs]: the only difference between these two would be when `watch` is true, call `watch_app*`

but they should both "install" first

File: crates/meroctl/src/cli/context/update.rs [hunk 4]
```diff
@@ -81,88 +85,58 @@ impl UpdateCommand {
...
-                    path,
-                    metadata.map(String::into_bytes),
+                    path.clone(),
+                    metadata.clone(),
                     &config.identity,
-                    executor_public_key,
                 )
                 .await?;
+                if self.watch {
+                    watch_app_and_update_context(
```
Comment (miraclx) [inline crates/meroctl/src/cli/context/update.rs:120]: this won't update the context unless the application changes

we should update and then watch, see `context create` for reference

**Review by miraclx — APPROVED** [review]

**Pull request merged**